frontend/common/src/components/TableOfContents/TableOfContentsAnchor.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { HTMLAttributes, MouseEventHandler } from "react";
+
+export interface TableOfContentsAnchor
+  extends HTMLAttributes<HTMLAnchorElement> {
+  id: string;
+}
+
+const TableOfContentsAnchor: React.FC<TableOfContentsAnchor> = ({
+  id,
+  children,
+}) => {
+  const [targetSection, setTargetSection] = React.useState<HTMLElement | null>(
+    null,
+  );
+
+  React.useEffect(() => {
+    const section = document.getElementById(id);
+    setTargetSection(section);
+  }, [id]);
+
+  const handleClick = (e: React.MouseEvent<HTMLAnchorElement>) => {
+    if (targetSection) {
+      e.preventDefault();
+      const offset = targetSection.offsetTop;
+      window.scroll({
+        top: offset,
+        behavior: "smooth",
+      });
+    }
+  };
+
+  return (
+    <a href={`#${id}`} data-h2-margin="b(bottom, xs)" onClick={handleClick}>
+      {children}
+    </a>
```

Comment:
I have to ask, why bother with a custom onClick? Do some browsers not handle the fragment href properly?

Reply:
I think it's for smooth scroll

Reply:
Yeah, this is only for the smooth scroll. If we do not care about that I can remove it and just use the `href` attribute.